src/Xtools/EditCounter.php

```diff
@@ -308,49 +317,117 @@ public function countBlocksReceived()
     }
 
     /**
-     * Get the length of the longest block the user received.
-     * @return int|bool Number of days or false if it could not be determined.
-     *                  If the longest duration is indefinite, -1 is returned.
+     * Get the length of the longest block the user received, in seconds.
+     * @return int Number of seconds or false if it could not be determined.
+     *   If the user is blocked, the time since the block is returned. If the block is
+     *   indefinite, -1 is returned. 0 if there was never a block.
      */
-    public function getLongestBlockDays()
+    public function getLongestBlockSeconds()
     {
-        if (isset($this->longestBlockDays)) {
-            return $this->longestBlockDays;
+        if (isset($this->longestBlockSeconds)) {
+            return $this->longestBlockSeconds;
         }
 
-        $blocks = $this->getBlocks('received'); // FIXME: make sure this is only called once
-        $this->longestBlockDays = false;
+        $blocks = $this->getBlocks('received', false);
+        $this->longestBlockSeconds = false;
 
-        foreach ($blocks as $block) {
-            $timestamp = strtotime($block['log_timestamp']);
+        // If there was never a block, the longest was zero seconds.
+        if (empty($blocks)) {
+            return 0;
+        }
 
-            // First check if the string is serialized, and if so parse it to get the block duration
-            if (@unserialize($block['log_params']) !== false) {
-                $parsedParams = unserialize($block['log_params']);
-                $durationStr = $parsedParams['5::duration'];
-            } else {
-                // Old format, the duration in English + block options separated by new lines
-                $durationStr = explode("\n", $block['log_params'])[0];
-            }
+        /**
+         * Keep track of the last block so we can determine the duration
+         * if the current block in the loop is an unblock.
+         * @var int[] [
+         *              Unix timestamp,
+         *              Duration in seconds (-1 if indefinite)
+         *            ]
+         */
+        $lastBlock = null;
+
+        foreach ($blocks as $index => $block) {
+            list($timestamp, $duration) = $this->parseBlockLogEntry($block);
+
+            if ($block['log_action'] === 'block') {
+                // This is a new block, so first see if the duration of the last
+                // block exceeded our longest duration. -1 duration means indefinite.
+                if ($lastBlock[1] > $this->longestBlockSeconds || $lastBlock[1] === -1) {
```

Comment:
The first time this loop executes, it's going to throw an undefined variable notice about `$lastBlock[1]`. Shouldn't `$lastBlock = [$timestamp, $duration];` happen earlier?

Reply:
We don't want to set the timestamp and duration yet, but you're right the initial declaration should be like `$lastBlock = [null, null]`